packages/core/src/defer/interfaces.ts

```diff
@@ -137,6 +174,34 @@ export enum DeferBlockState {
   Error = 3,
 }
 
+/**
+ * Describes the defer trigger type.
+ *
+ * @publicApi
+ */
+export enum Trigger {
+  Idle = 0,
+
+  Immediate = 1,
+
+  Viewport = 2,
+
+  Interaction = 3,
+
+  Hover = 4,
+
+  Timer = 5,
+
+  When = 6,
+
+  Never = 7,
+}
+
+export interface HydrateTrigger {
+  trigger: Trigger;
+  delay?: number;
+}
```

Comment:
It'd be also great to add JsDoc to this type and its fields.
```suggestion
export interface HydrateTriggerDetails {
  trigger: Trigger;
  delay?: number;
}
```